### PR TITLE
stress test + audit bug fix

### PR DESCRIPTION
**Pull request by adecaro**
This PR brings the following:
- An integration test that can be run locally for stress tests;
- A fix to the auditor lock strategy that avoids deadlocks that was discovered during the stress test;

### CODE REVIEW

File: integration/token/fungible/dlogstress/support/worker.go [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+		workers:   make([]*Worker, numWorkers),
+		taskQueue: make(chan Task, numWorkers),
+		wg:        sync.WaitGroup{},
+		shutdown:  make(chan struct{}, 1),
+	}
+
+	for i := 0; i < numWorkers; i++ {
+		worker := &Worker{id: i + 1, taskQueue: pool.taskQueue, wg: &pool.wg}
+		worker.start(&pool.wg)
+		pool.workers = append(pool.workers, worker)
```
Comment (alexandrosfilios) [inline integration/token/fungible/dlogstress/support/worker.go:55]: `pool.workers[I] = worker`
or above
`workers: make([]*Worker, 0, numWorkers)`

File: integration/token/fungible/dlogstress/support/worker.go [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+func NewPool(numWorkers int) *Pool {
+	pool := &Pool{
+		workers:   make([]*Worker, numWorkers),
+		taskQueue: make(chan Task, numWorkers),
+		wg:        sync.WaitGroup{},
+		shutdown:  make(chan struct{}, 1),
+	}
+
+	for i := 0; i < numWorkers; i++ {
+		worker := &Worker{id: i + 1, taskQueue: pool.taskQueue, wg: &pool.wg}
```
Comment (alexandrosfilios) [inline integration/token/fungible/dlogstress/support/worker.go:53]: Do we really need the worker? We could do directly
```
for i := 0; i < numWorkers; i++ {
go func() {
 defer wg.Done()
 for {
   if case task, ok := <-w.taskQueue; ok {task()} else {return}
 }
}()
} 
```

File: token/services/auditdb/db.go [hunk 3]
```diff
@@ -253,15 +259,15 @@ func (d *DB) GetTokenRequest(txID string) ([]byte, error) {
 // AcquireLocks acquires locks for the passed anchor and enrollment ids.
 // This can be used to prevent concurrent read/write access to the audit records of the passed enrollment ids.
 func (d *DB) AcquireLocks(anchor string, eIDs ...string) error {
-	dedup := deduplicate(eIDs)
-	logger.Debugf("Acquire locks for [%s:%v] enrollment ids", anchor, dedup)
+	dedup := deduplicateAndSort(eIDs)
```
Comment (alexandrosfilios) [inline token/services/auditdb/db.go:262]: You could use existing methods
```
dedup := collections.NewSet(eIDs...).ToSlice()
slices.Sort(dedup)
```

File: token/services/auditdb/db.go [hunk 4]
```diff
@@ -121,6 +122,11 @@ type Wallet interface {
 	TMS() *token.ManagementService
 }
 
+type LockEntry struct {
```
Comment (alexandrosfilios) [inline token/services/auditdb/db.go:125]: What do we need this for?

Comment (adecaro) [reply token/services/auditdb/db.go:125]: good catch, we don't need that

File: token/services/auditdb/db.go (hunk 3 above)
Comment (adecaro) [reply token/services/auditdb/db.go:262]: Done. I have added also a comment to explain the problem we want to solve.

File: integration/token/fungible/dlogstress/support/worker.go (hunk 1 above)
Comment (adecaro) [reply integration/token/fungible/dlogstress/support/worker.go:55]: I will go with make([]*Worker, 0, numWorkers)

File: integration/token/fungible/tests.go [hunk 5]
```diff
@@ -973,3 +975,63 @@ func TestMaliciousTransactions(net *integration.Infrastructure) {
...
+					logger.Errorf("caught panic during issue: %v", r)
+				}
+			}()
+			IssueCash(network, "", "StressCoin", 1, "alice", auditor, true, "issuer")
+			IssueCash(network, "", "StressCoin", 1, "bob", auditor, true, "issuer")
+		})
+	}()
+
+	// let issue enough tokens
+	time.Sleep(1 * time.Minute)
```
[the remark this answers is not in the document]
Comment (adecaro) [reply integration/token/fungible/tests.go:1003]: I just need to wait that enough issues have been generated. I do not need more.

**Review by alexandrosfilios — APPROVED** [review]
LGTM

**Pull request merged**